src/h5web/visualizations/heatmap/AxisGrid.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react';
+import { Dom, useThree, useFrame } from 'react-three-fiber';
+import { scaleLinear } from 'd3-scale';
+import IndexAxis from './IndexAxis';
+import styles from './HeatmapVis.module.css';
+
+interface AxisDomains {
+  left: [number, number];
+  bottom: [number, number];
+}
+
+interface Props {
+  dims: [number, number];
+  axisOffsets: [number, number];
+}
+
+function AxisGrid(props: Props): JSX.Element {
+  const { dims, axisOffsets } = props;
+  const [rows, cols] = dims;
+  const [leftAxisWidth, bottomAxisHeight] = axisOffsets;
+
+  const { camera, size } = useThree();
+  const { width, height } = size;
+
+  const [domains, setDomains] = useState<AxisDomains>({
+    left: [0, rows],
+    bottom: [0, cols],
+  });
+
+  // Axis bounds in R3F camera coordinates
+  const leftBounds: [number, number] = [
+    -height / 2 + bottomAxisHeight,
+    height / 2,
+  ];
```

Comment:
You can put `as const` here instead of typing the constant.

Reply:
I wish but Typescript complains when setting it as a `d3-scale` domain...